header-footer-grid/Core/Components/EddCartIcon.php

```diff
@@ -0,0 +1,217 @@
+<?php
+/**
+ * Easy Digital Downloads Component class for Header Footer Grid.
+ *
+ * Author:          Uriahs Victor
+ * Created on:      06/01/2022 (d/m/y)
+ *
+ * @package HFG
+ */
+
+namespace HFG\Core\Components;
+
+use HFG\Main;
+use HFG\Core\Settings\Manager as SettingsManager;
+use Neve\Core\Styles\Dynamic_Selector;
+
+
+/**
+ * Class Yoast_Breadcrumbs
+ *
+ * @package Neve_Pro\Modules\Header_Footer_Grid\Components
+ */
+class EddCartIcon extends Abstract_Component {
+
+	const COMPONENT_ID   = 'edd_cart';
+	const ICON_SELECTOR  = 'icon_selector';
+	const CART_LABEL     = 'cart_label';
+	const SIZE_ID        = 'icon_size';
+	const LABEL_SIZE_ID  = 'label_size';
+	const COLOR_ID       = 'color';
+	const HOVER_COLOR_ID = 'hover_color';
+
+
+	/**
+	 * Breadcrumbs constructor.
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 */
+	public function init() {
+		$this->set_property( 'label', __( 'Easy Digital Downloads Cart', 'neve' ) );
+		$this->set_property( 'id', self::COMPONENT_ID );
+		$this->set_property( 'width', 1 );
+		$this->set_property( 'default_selector', '.builder-item--' . $this->get_id() );
+	}
+
+	/**
+	 * Called to register component controls.
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 */
+	public function add_settings() {
+
+		do_action( 'nv_edd_cart_icon_component_controls' );
```

Comment:
Let's drop this hook and instead use the `Abstract_Component` hook that is already defined here: 

https://github.com/codeinwp/neve/blob/b1a79187d586206be44869b9321e1dbc2bbf0ac0/header-footer-grid/Core/Components/Abstract_Component.php#L547